src/apps/replication/meta_server/server_state.h

```diff
@@ -163,6 +171,11 @@ class server_state :
     };
 
     mutable zlock                         _pending_requests_lock;
-    std::map<uint64_t, storage_work_item> _pending_requests;
+    // because ballots of different gpid may conflict, we separate items by gpid
+    // app_id -> pidx -> <ballot, item>
+    std::vector< std::vector< std::map<uint64_t, storage_work_item> > > _pending_requests;
```

Comment:
improved in https://github.com/qinzuoyan/rDSN/commit/eb135f225643b099bebcfd6a1f47239eb8d87d84